k8s-install/scripts/install_cni_test.go

```diff
@@ -0,0 +1,208 @@
+package scripts_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/projectcalico/felix/fv/containers"
+
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+var secretFile = "tmp/serviceaccount/token"
+
+// runCniContainer will run install-cni.sh.
+// TODO: We should be returning an error if the command fails, there currently is
+// not a way to get that from the container package.
+func runCniContainer(extraArgs ...string) error {
+	// Get the CWD for mounting directories into container.
+	cwd, err := os.Getwd()
+	if err != nil {
+		Fail("Could not get CWD")
+	}
+
+	// Assemble our arguments.
+	args := []string{
+		"--privileged",
+		"-e", "SLEEP=false",
+		"-e", "KUBERNETES_SERVICE_HOST=127.0.0.1",
+		"-e", "KUBERNETES_SERVICE_PORT=8080",
+		"-v", cwd + "/tmp/bin:/host/opt/cni/bin",
+		"-v", cwd + "/tmp/net.d:/host/etc/cni/net.d",
+		"-v", cwd + "/tmp/serviceaccount:/var/run/secrets/kubernetes.io/serviceaccount",
+	}
+	args = append(args, extraArgs...)
+	args = append(args, "calico/cni:latest", "/install-cni.sh")
+
+	c := containers.Run("cni", args...)
+	for !c.Stopped() {
```

Comment:
It's scary to loop forever here - could we instead use a gomega `Eventually` statement?

https://onsi.github.io/gomega/#eventually